src/rgw/rgw_gc.cc

```diff
@@ -65,7 +65,64 @@ int RGWGC::tag_index(const string& tag)
   return rgw_shards_mod(XXH64(tag.c_str(), tag.size(), seed), max_objs);
 }
 
-int RGWGC::send_chain(cls_rgw_obj_chain& chain, const string& tag)
+int RGWGC::send_split_chain(const cls_rgw_obj_chain& chain, const std::string& tag)
+{
+  constexpr int FACTOR_INCREASE = 3; //approximate value by which total size of chain, tag and time increases after encoding into bufferlist
+  ldpp_dout(this, 20) << "RGWGC::send_split_chain - tag is: " << tag << dendl;
+
+  if (cct->_conf->osd_max_write_size) {
+    cls_rgw_obj_chain broken_chain;
+    bool chain_sent = false;
+    long unsigned int tag_size = tag.length(), expiration_time_size = sizeof(ceph::real_time);
```

Comment:
can these safely be auto? otherwise, I would recomment explicit stdint types:  uint64_t, &c